javalin/src/main/java/io/javalin/core/util/Headers.kt

```diff
@@ -0,0 +1,144 @@
+        if (includeSubdomains) {
+            headers[Header.STRICT_TRANSPORT_SECURITY] = "max-age=" + duration.seconds.toString() + " ; includeSubDomains"
+        } else {
+            headers[Header.STRICT_TRANSPORT_SECURITY] = "max-age=" + duration.seconds
+        }
+    }
+
+    // X-Frame-Options: deny | sameorigin | allow-from: DOMAIN
+    enum class XFrameOptions {
+        DENY, SAMEORIGIN;
+
+        fun headerValue(): String {
+            return name.toLowerCase(Locale.ROOT)
+        }
+    }
+
+    fun xFrameOptions(xFrameOptions: XFrameOptions) {
+        headers[Header.X_FRAME_OPTIONS] = xFrameOptions.headerValue()
+    }
+
+    fun xFrameOptions(domain: String) {
+        headers[Header.X_FRAME_OPTIONS] = "allow-from: $domain"
+    }
+
+    // X-Content-Type-Options: nosniff
+    fun xContentTypeOptionsNoSniff() {
+        headers[Header.X_CONTENT_TYPE_OPTIONS] = "nosniff"
+    }
+
+    // Content-Security-Policy: String... + JAVADOC
+    fun contentSecurityPolicy(contentSecurityPolicy: String) {
+        headers[Header.CONTENT_SECURITY_POLICY] = contentSecurityPolicy
+    }
+
+    // X-Permitted-Cross-Domain-Policies: none | master-only | by-content-type | by-ftp-filename | all
+    enum class CrossDomainPolicy {
+        NONE, MASTER_ONLY, BY_CONTENT_TYPE, BY_FTP_FILENAME, ALL;
+
+        fun headerValue(): String {
+            return name.toLowerCase(Locale.ROOT).replace("_", "-")
+        }
+    }
+
+    fun xPermittedCrossDomainPolicies(policy: CrossDomainPolicy) {
+        headers[Header.X_PERMITTED_CROSS_DOMAIN_POLICIES] = policy.headerValue()
+    }
+
+    // Referrer-Policy: no-referrer | no-referrer-when-downgrade | origin | origin-when-cross-origin | same-origin | strict-origin | strict-origin-when-cross-origin | unsafe-url
+    enum class ReferrerPolicy {
+        NO_REFERRER, NO_REFERRER_WHEN_DOWNGRADE, ORIGIN, ORIGIN_WHEN_CROSS_ORIGIN, SAME_ORIGIN, STRICT_ORIGIN, STRICT_ORIGIN_WHEN_CROSS_ORIGIN, UNSAFE_URL;
+
+        fun headerValue(): String {
+            return name.toLowerCase(Locale.ROOT).replace("_", "-")
+        }
+    }
+
+    fun referrerPolicy(policy: ReferrerPolicy) {
+        headers[Header.REFERRER_POLICY] = policy.headerValue()
+    }
+
+    // Clear-Site-Data: "cache" | "cookies" | "storage" | "executionContexts" | "*"
+    enum class ClearSiteData {
+        CACHE, COOKIES, STORAGE, EXECUTION_CONTEXTS, ANY;
+
+        fun headerValue(): String {
+            if (this == ANY) {
+                return "\"*\""
+            } else if (this == EXECUTION_CONTEXTS) {
+                return "\"executionContexts\""
+            }
+            return "\"" + name.toLowerCase(Locale.ROOT) + "\""
+        }
+    }
+
+    fun clearSiteData(vararg data: ClearSiteData) {
+        val value: String = Arrays.stream(data)
+                .map { obj: ClearSiteData -> obj.headerValue() }
+                .collect(Collectors.joining(","))
```

Comment:
I think you should be able to get away with something like `data.map{}.joinToString(",")` ?